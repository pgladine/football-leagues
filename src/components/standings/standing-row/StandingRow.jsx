import React from 'react'
import './standing-row.css'

export default function StandingRow({ position, teamCrest, teamName, points, playedGames, won, draw, lost, goalsFor, goalsAgainst }) {
  return (
    <div className='standing-container'>
      <div className='home-team-info'>
        <span>{position}</span>
        <img src={teamCrest} alt="escudo" />
        <span>{teamName}</span>
      </div>
      <div className='team-numbers'>
        <span className='total-points'>{points}</span>
        <span>{playedGames}</span>
        <span>{won}</span>
        <span>{draw}</span>
        <span>{lost}</span>
        <span>{goalsFor}</span>
        <span>{goalsAgainst}</span>
      </div>
    </div>
  )
}