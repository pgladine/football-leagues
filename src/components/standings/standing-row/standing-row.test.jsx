import { render, cleanup, screen } from "@testing-library/react";
import StandingRow from './StandingRow'

describe('<StandingRow/>', () => {
  afterEach(cleanup)

  it('should render the data with the right props', () => {
    // Given
    const teamData = {
      position: 1,
      teamCrest: 'https://teamCrest.url',
      teamName: 'teamName',
      points: 15,
      playedGames: 6,
      won: 5,
      draw: 0,
      lost: 1,
      goalsFor: 20,
      goalsAgainst: 6,
    }
    // When
    render(<StandingRow {...teamData} />)

    // Then
    expect(screen.getByAltText('escudo')).toHaveAttribute('src', teamData.teamCrest)
    expect(document.body).toMatchSnapshot()
  })
})