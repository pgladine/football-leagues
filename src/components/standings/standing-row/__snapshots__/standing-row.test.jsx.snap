// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<StandingRow/> > should render the data with the right props 1`] = `
<body>
  <div>
    <div
      class="standing-container"
      data-testid="standing-row"
    >
      <div
        class="home-team-info"
      >
        <span>
          1
        </span>
        <img
          alt="escudo"
          src="https://teamCrest.url"
        />
        <span>
          teamName
        </span>
      </div>
      <div
        class="team-numbers"
      >
        <span
          class="total-points"
        >
          15
        </span>
        <span>
          6
        </span>
        <span>
          5
        </span>
        <span>
          0
        </span>
        <span>
          1
        </span>
        <span>
          20
        </span>
        <span>
          6
        </span>
      </div>
    </div>
  </div>
</body>
`;
